src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy

```diff
@@ -0,0 +1,89 @@
+package com.skalashynski.spring.springboot.web.controller
+
+import com.skalashynski.spring.springboot.Application
+import com.skalashynski.spring.springboot.DatabaseSpecification
+import com.skalashynski.spring.springboot.entity.Student
+import com.skalashynski.spring.springboot.util.ResourceUtils
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.http.ResponseEntity
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.ContextConfiguration
+import spock.lang.Unroll
+
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@SpringBootTest(
+        classes = Application.class,
+        webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT
+)
+@ActiveProfiles(value = "test")
+@ContextConfiguration
+class StudentControllerTest extends DatabaseSpecification {
+
+    private TestRestTemplate restTemplate = new TestRestTemplate();
+    private HttpHeaders headers = new HttpHeaders();
+
+    static String setupContent
+    static String cleanupContent
+
+//  выполняется 1 раз перед всеми тестами (подгрузка ресурсов)
+    def setupSpec() {
+        cleanupContent = ResourceUtils.getResourceContents('cleanup.sql')
+        setupContent = ResourceUtils.getResourceContents('data.sql')
+    }
+
+//  выполняется перед каждым тестом
+    @SuppressWarnings("unused")
+    def setup() {
+        //used 1 year expiration token
+        headers.add("Authorization"
+                , "Bearer eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiJpZF92YWx1ZSIsImlhdCI6MTYzMTcxNzM2MiwiYXV0aG9yaXRpZXMiOlt7ImF1dGhvcml0eSI6IlJPTEVfU1RVREVOVCJ9XSwic3ViIjoiU2VyZ2lLMyIsImlzcyI6IlNwcmluZ19EZW1vX0FQSV9TZXJ2aWNlIiwiZXhwIjoxNjYzMjUzMzYyfQ.D4EVYJu6Cyb82M7vs1QphF11pQYWIgLo57BbHbfgeKQ")
+        sql.execute(setupContent)
+    }
+
+//  выполняется после каждого теста (очистка таблицы)
+    @SuppressWarnings("unused")
+    def cleanup() {
+        sql.execute(cleanupContent)
+    }
+
+//  выполняется после всех тестов 1 раз (освобождает ресурсы)
+//    def cleanupSpec() {
+//    }
+
+    @Unroll
+    def void "Student get by id"(Long id) {
+        when:
```

Comment:
For Spock testing use please #field in the method name
https://www.jetbrains.com/idea/guide/tutorials/writing-tests-with-spock/data-tables/
16:00

Reply:
Corrected according to recommendations